src/accelerate/accelerator.py

```diff
@@ -316,10 +312,24 @@ def __init__(
             mixed_precision = (
                 os.environ.get("ACCELERATE_MIXED_PRECISION", "no") if mixed_precision is None else mixed_precision
             )
-            for plugin in deepspeed_plugin:
-                plugin.set_mixed_precision(mixed_precision)
-            # The first plugin is always the active one
-            deepspeed_plugin[0].enable()
+            if not isinstance(deepspeed_plugin, (tuple, list, dict)):
+                deepspeed_plugin.set_mixed_precision(mixed_precision)
+                deepspeed_plugin.enable()
+            else:
+                # Can either be a list or a dict at this point:
+                # list if we haven't initialized the plugins yet
+                # dict if we have
+                for plugin in deepspeed_plugin:
+                    if not isinstance(plugin, DeepSpeedPlugin):
+                        plugin = deepspeed_plugin[plugin]
+                    plugin.set_mixed_precision(mixed_precision)
+                if isinstance(deepspeed_plugin, (list, tuple)):
+                    # The first plugin is always the active one
+                    deepspeed_plugin[0].enable()
```

Comment:
aren't you missing `enable` in the case of dict?

Reply:
My thoughts here are:

Users set the name as part of the config object itself. Rather than having many ways to do one thing, I think it'd make sense to have users just set the name only during the config, and users still passing a list/tuple. They're already passing in a configured DS plugin anyways.

Reply:
Okay I'm convinced the other way, you have to specify it as a `dict`, you can't set it as part of the plugin config :)